DigitalLearningSolutions.Web.Tests/ViewModels/LearningMenu/CourseCompletionViewModelTests.cs

```diff
@@ -0,0 +1,384 @@
+        }
+
+        [Test]
+        public void CourseCompletion_FinaliseText_should_be_null_when_completed_is_null()
+        {
+            // Given
+            var expectedCourseCompletion = CourseCompletionHelper.CreateDefaultCourseCompletion(completed: null);
+
+            // When
+            var courseCompletionViewModel = new CourseCompletionViewModel(expectedCourseCompletion);
+
+            // Then
+            courseCompletionViewModel.FinaliseText.Should().BeNull();
+        }
+
+        [Test]
+        public void CourseCompletion_FinaliseAriaLabel_should_be_null_when_completed_is_null()
+        {
+            // Given
+            var expectedCourseCompletion = CourseCompletionHelper.CreateDefaultCourseCompletion(completed: null);
+
+            // When
+            var courseCompletionViewModel = new CourseCompletionViewModel(expectedCourseCompletion);
+
+            // Then
+            courseCompletionViewModel.FinaliseAriaLabel.Should().BeNull();
+        }
+
+        [Test]
+        public void CourseCompletion_FinaliseText_should_be_certificate_when_course_is_assessed()
+        {
+            // Given
+            var expectedCourseCompletion = CourseCompletionHelper.CreateDefaultCourseCompletion(
+                completed: DateTime.Now,
+                isAssessed: true
+            );
+
+            // When
+            var courseCompletionViewModel = new CourseCompletionViewModel(expectedCourseCompletion);
+
+            // Then
+            courseCompletionViewModel.FinaliseText.Should().Be("Certificate");
+        }
+
+        [Test]
+        public void CourseCompletion_FinaliseAriaLabel_should_be_certificate_when_course_is_assessed()
+        {
+            // Given
+            var expectedCourseCompletion = CourseCompletionHelper.CreateDefaultCourseCompletion(
+                completed: DateTime.Now,
+                isAssessed: true
+            );
+
+            // When
+            var courseCompletionViewModel = new CourseCompletionViewModel(expectedCourseCompletion);
+
+            // Then
+            courseCompletionViewModel.FinaliseAriaLabel.Should().Be("View or print certificate");
+        }
+
+        [Test]
+        public void CourseCompletion_FinaliseText_should_be_evaluate_when_course_is_not_assessed_and_not_evaluated()
+        {
+            // Given
+            var expectedCourseCompletion = CourseCompletionHelper.CreateDefaultCourseCompletion(
+                completed: DateTime.Now,
+                evaluated: null,
+                isAssessed: false
+            );
+
+            // When
+            var courseCompletionViewModel = new CourseCompletionViewModel(expectedCourseCompletion);
+
+            // Then
+            courseCompletionViewModel.FinaliseText.Should().Be("Evaluate");
+        }
+
+        [Test]
+        public void CourseCompletion_FinaliseAriaLabel_should_be_evaluate_when_course_is_not_assessed_and_not_evaluated()

```

Comment:
What about if the course is assessed and not evaluated? Do we have a test for that case? 🙂

Reply:
This corresponds to the "Complete evaluation and access certificate" button that I had in the design at one point, which now just says "Certificate", based on the implementation of this button in the Completed courses learning portal page. I've added two tests to cover this in adf3f44 🙂